packages/core/src/test/credentials/sharedCredentials.test.ts

```diff
@@ -17,6 +19,7 @@ describe('sharedCredentials', function () {
         // Make a temp folder for all these tests
         // Stick some temp credentials files in there to load from
         tempFolder = await makeTemporaryToolkitFolder()
+        sinon.stub(pathUtils, 'isValidPath').returns(true)
```

Comment:
stubbing isValidPath in various tests that happen to current hit the `getHomeDirectory()` codepath, means that future tests might run into the same issue.

Reply:
Given that in unit tests we are validating an actual path, indeed it will require stubbing in future as well. What would be your recommendation?

Reply:
The [change](https://github.com/aws/aws-toolkit-vscode/pull/4901#discussion_r1667731918) that I pushed avoids this problem by only validating at startup (when `initUserHomeDir()` is called).